pkg/kubernetes/details.js

```diff
@@ -11,30 +11,111 @@ define([
     var _ = cockpit.gettext;
 
     var k8client =  kubernetes.k8client();
-    var adjust_btn = $('#adjust-entity-button');
-    var adjust_entity_dlg = $('#adjust-entity-dialog');
+    var adjust_btn = $('#adjust-rc-button');
+    var adjust_rc_dlg = $('#adjust-rc-dialog');
+    var adjust_route_btn = $('#adjust-route-button');
     var delete_btn = $('#delete-entity-button');
     var delete_entity_dlg = $('#delete-entity-dialog');
     var delete_pod_dlg = $('#delete-pod-dialog');
     var delete_pod_btn = $('#delete-pod-button');
+    var adjust_route_dlg = $('#adjust-route-dialog');
 
     delete_entity_dlg.on('show.bs.modal', function(e) {
         var key = $(e.relatedTarget).attr("data-key");
         var entity = k8client.objects[key];
         delete_entity_dlg.find('.modal-body').text(cockpit.format(_("Delete $0 $1?"), entity.kind, entity.metadata.name));
 
-        delete_btn.on('click', function() {
+        delete_btn.off('click').on('click', function() {
             var promise = k8client.remove(entity.metadata.selfLink);
             delete_entity_dlg.dialog("promise", promise);
         });
     });
 
-    adjust_entity_dlg.on('show.bs.modal', function(e) {
+    adjust_route_dlg.on('show.bs.modal', function(e) {
         var key = $(e.relatedTarget).attr("data-key");
         var entity = k8client.objects[key];
-        adjust_entity_dlg.find('#replica-count').val(entity.spec.replicas);
+        adjust_route_dlg.find('#host-value').val(entity.spec.host);
 
-        adjust_btn.on('click', function() {
+        adjust_route_btn.off('click').on('click', function() {
+
+            var deferred = $.Deferred();
+            adjust_route_dlg.dialog("failure", null);
+            adjust_route_dlg.dialog("wait", deferred.promise());
```

Comment:
I don't understand what the point of this external deferred wrapper is? Can't the promise returned from update_host be passed directly to ```adjust_route_dlg.dialog("promise", promise)``` ?